### PR TITLE
Use registry passed in from DatabaseOptions in newDatabase.

### PR DESCRIPTION
**Pull request by athum**
GODRIVER-1325

Change-Id: I6e63baf092a796c1e9ca01e4c84c8cac6f6af4ca

### CODE REVIEW

**Comment by athum** [comment]
evergreen: https://evergreen.mongodb.com/version/5d939fc13627e009b36f8d02

File: mongo/database.go [hunk 1]
```diff
@@ -58,13 +58,18 @@ func newDatabase(client *Client, name string, opts ...*options.DatabaseOptions)
...
+	}
+
 	db := &Database{
 		client:         client,
 		name:           name,
 		readPreference: rp,
 		readConcern:    rc,
 		writeConcern:   wc,
-		registry:       client.registry,
+		registry:       rg,
```
Comment (jyemin) [inline mongo/database.go]: If you revert this change does the new test still pass?

Comment (athum) [reply mongo/database.go]: Yes, a method has been added to compare registries in the test file that would fail without the newest change.

**Comment by athum** [comment]
evergreen: https://evergreen.mongodb.com/version/5d93bd1d3e8e865026379ab1

File: mongo/database.go (hunk 1 above)
Comment (jyemin) [reply mongo/database.go]: Yes, looks better now!

File: mongo/main_test.go [hunk 2]
```diff
@@ -24,6 +25,10 @@ func TestMain(m *testing.M) {
 	assert.RegisterOpts(reflect.TypeOf(&Client{}), cmp.Comparer(func(c1, c2 *Client) bool {
 		return c1 == c2
 	}))
+	assert.RegisterOpts(reflect.TypeOf(&bsoncodec.Registry{}), cmp.Comparer(func(r1, r2 *bsoncodec.Registry) bool {
```
Comment (jyemin) [inline mongo/main_test.go]: Nice!
